zeroex/ordervalidator/order_validator_js.go

```diff
@@ -43,3 +54,47 @@ func (s RejectedOrderStatus) JSValue() js.Value {
 		"message": s.Message,
 	})
 }
+
+const computeOptimalChunkBatchSize = 3
+
+// computeOptimalChunkSizes splits the signedOrders into chunks where the payload size of each chunk
+// is beneath the maxRequestContentLength. It does this by implementing a greedy algorithm which ABI
+// encodes signedOrders one at a time until the computed payload size is as close to the
+// maxRequestContentLength as possible.
+func (o *OrderValidator) computeOptimalChunkSizes(signedOrders []*zeroex.SignedOrder) []int {
+	chunkSizes := []int{}
+
+	payloadLength := jsonRPCPayloadByteLength
+	nextChunkSize := 0
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		batchIdx := 0
+		for _, signedOrder := range signedOrders {
+			if batchIdx%computeOptimalChunkBatchSize == 2 {
+				time.Sleep(sleepTime)
+			}
+			encodedSignedOrderByteLength, _ := o.computeABIEncodedSignedOrderByteLength(signedOrder)
+			if payloadLength+encodedSignedOrderByteLength < o.maxRequestContentLength {
+				payloadLength += encodedSignedOrderByteLength
+				nextChunkSize++
+			} else {
+				if nextChunkSize == 0 {
+					// This case should never be hit since we enforce that EthereumRPCMaxContentLength >= maxOrderSizeInBytes
+					log.WithField("signedOrder", signedOrder).Panic("EthereumRPCMaxContentLength is set so low, a single 0x order cannot fit beneath the payload limit")
+				}
+				chunkSizes = append(chunkSizes, nextChunkSize)
+				nextChunkSize = 1
+				payloadLength = jsonRPCPayloadByteLength + encodedSignedOrderByteLength
+			}
+			batchIdx = batchIdx + 1
+		}
+		wg.Done()
```

Comment:
Can you move this `wg.Done()` call to the top of the goroutine and use `defer`? This guarantees we always call it and prevents deadlock.